docs/using_python_client_with_hazelcast.rst

```diff
@@ -1572,22 +1572,114 @@ Query parameters have the following benefits:
 Querying JSON Objects
 ~~~~~~~~~~~~~~~~~~~~~
 
-To query JSON objects, you should create an explicit mapping using the
+In Hazelcast, the SQL service supports the following ways of working with
+JSON data:
+
+- ``json``: Maps JSON data to a single column of ``JSON`` type where you can
+  use `JsonPath
+  <https://docs.hazelcast.com/hazelcast/latest/sql/working-with-json#querying-json>`__
+  syntax to query and filter it, including nested levels.
+- ``json-flat``: Maps JSON top-level fields to columns with non-JSON types
+  where you can query only top-level keys.
+
+**json**
+
+To query ``json`` objects, you should create an explicit mapping using the
 `CREATE MAPPING
 <https://docs.hazelcast.com/hazelcast/latest/sql/create-mapping.html>`__
 statement, similar to the example above.
 
 For example, this code snippet creates a mapping to a new map called
-``json_employees``, which stores the JSON values ``name`` and ``salary`` and
-query it:
+``json_employees``, which stores the JSON values as ``HazelcastJsonValue``
+objects and queries it using nested fields, which is not possible with the
+``json-flat`` type:
 
 .. code:: python
 
-    client = hazelcast.HazelcastClient()
+    client.sql.execute(
+        """
+    CREATE MAPPING IF NOT EXISTS json_employees
```

Comment:
CREATE OR REPLACE MAPPING would be better in case the user has a mapping created using an older example.